src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -75,38 +75,29 @@ private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnost
             return document;
         }
 
-        //whole invocation starting from "builder"
+        // builder.Host.ConfigureLogging(builder => builder.AddJsonConsole());
         var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); 
 
-        if (diagnosticTarget is InvocationExpressionSyntax originalInvocation)
+        if (diagnosticTarget is InvocationExpressionSyntax invocation)
         {
-            // if builder.{Host or WebHost}.{method_name} or builder.{Host or WebHost}
-            // is not of type MemberAccessExpressionSyntax, return original code
-            if (originalInvocation.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr
-                || hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax subExpression)
+            // No modification are made if the invocation isn't accessing a method on `builder.Host` or `builder.WebHost`.
+            if (invocation.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr
+                || hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax configureMethodOnHostAccessExpr)
             {
                 return document;
             }
 
-            subExpression = subExpression.WithName(identifierMethod);
+            configureMethodOnHostAccessExpr = configureMethodOnHostAccessExpr.WithName(identifierMethod);
             var indentation = hostBasedInvocationMethodExpr.GetLeadingTrivia();
-            // replace Host/WebHost with identifierMethod
-            hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(subExpression)
-                .NormalizeWhitespace().WithLeadingTrivia(indentation); 
-
-            if (originalInvocation.ArgumentList.Arguments.Count == 0)
-            {
-                return document;
-            }
 
-            var initArgumentExpr = originalInvocation.ArgumentList.Arguments.SingleOrDefault().Expression;   // builder => { }
+            // builder.Host.ConfigureLogging => builder.Logging
+            // builder.WebHost.ConfigureServices => builder.Services
+            hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(configureMethodOnHostAccessExpr)
+                .NormalizeWhitespace().WithLeadingTrivia(indentation); 
 
-            if (initArgumentExpr == null)
-            {
-                return document;
-            }
+            var initArgumentExpr = invocation.ArgumentList.Arguments.SingleOrDefault().Expression;   // builder => { }
```

Comment:
The ordering is a little weird now. Here we use `SingleOrDefault()` to get a single element from the list but we validate the size of the list afterwards. You can do the `SingleOrDefault` check in the if-statement.